src/components/global/header/header.tsx

```diff
@@ -2,25 +2,41 @@
 import Link from 'next/link'
 import { useWeb3Modal } from '@web3modal/wagmi/react'
 import { useAccount } from 'wagmi'
-import { Fragment } from 'react'
-import { Disclosure, Menu, Transition } from '@headlessui/react'
+import { Fragment, useEffect, useRef, useState } from 'react'
+import { Disclosure, Popover, Transition } from '@headlessui/react'
 import { Bars3Icon, XMarkIcon } from '@heroicons/react/24/outline'
 
 import * as utils from '@/utils'
 import * as hooks from '@/hooks'
 
 import peanut_logo from '@/assets/peanutman-logo.svg'
-import triangle_svg from '@/assets/icons/triangle.svg'
-
-function classNames(...classes: any) {
-    return classes.filter(Boolean).join(' ')
-}
 
 export function Header({ showMarquee = true }: { showMarquee?: boolean }) {
     const { address, isConnected } = useAccount()
-    const gaEventTracker = hooks.useAnalyticsEventTracker('header')
-
     const { open: web3modalOpen } = useWeb3Modal()
+    const buttonRef = useRef(null)
+    const [openState, setOpenState] = useState(false)
+
+    const onHover = (open: any, action: string) => {
+        if ((!open && !openState && action === 'onMouseEnter') || (open && openState && action === 'onMouseLeave')) {
+            setOpenState((openState) => !openState)
+            //@ts-ignore
+            buttonRef?.current?.click()
+        }
+    }
+
+    const handleClickOutside = (event: any) => {
+        //@ts-ignore
+        if (buttonRef.current && !buttonRef.current.contains(event.target)) {
```

Comment:
ts ignore? :question:

Reply:
yes, ts doesnt like the any type. thats why I used tsignore. But the <HTMLButtonElement> definition you suggested fixes this, ty for that